odl/contrib/fom/supervised/supervised_fom.py

```diff
@@ -0,0 +1,575 @@
+
+    or, in normalized form
+
+    .. math::
+        \\bigg \\lvert \\frac{ \\overline{f} - \\overline{g} }
+                                 {\\overline{f} + \\overline{g}} \\bigg
+                                                                 \\rvert.
+
+    where
+
+    .. math::
+        \\overline{f} := \\frac{1}{\|1_\Omega\|_1} \\int_\Omega f dx,
+
+    and
+
+    .. math::
+        \\overline{g} := \\frac{1}{\|1_\Omega\|_1} \\int_\Omega g dx.
+
+    The normalized FOM takes its values in [0, 1], with higher correspondance
+    at lower FOM value.
+
+    Parameters
+    ----------
+    data : `FnBaseVector`
+        Input data or reconstruction.
+    ground_truth : `FnBaseVector`
+        Reference to compare ``data`` to.
+    mask : `FnBaseVector`
+        Binary mask to define ROI in which FOM evaluation is performed.
+    normalized  : bool
+        Boolean flag to switch between unormalized and normalized FOM.
+
+    Returns
+    -------
+    fom : float
+        Scalar (float) indicating difference in mean value between
+        ``data`` and ``ground_truth``. In normalized form the FOM takes
+        values in [0, 1], with higher correspondance at lower FOM value.
+    """
+
+    import numpy as np
+
+    l1_norm = odl.solvers.L1Norm(data.space)
+
+    if mask:
+        data = data * mask
+        ground_truth = ground_truth * mask
+
+    # Identify positive part of data
+    pos_data = data.copy()
+    pos_data = pos_data.asarray()
+    pos_data[pos_data < 0] = 0
+    pos_data = data.space.element(pos_data)
+
+    # Identify negative part of data
+    neg_data = data.copy()
+    neg_data = neg_data.asarray()
+    neg_data[neg_data > 0] = 0
+    neg_data = data.space.element(neg_data)
+
+    # Identify positive part of ground_truth
+    pos_ground_truth = ground_truth.copy()
+    pos_ground_truth = pos_ground_truth.asarray()
+    pos_ground_truth[pos_ground_truth < 0] = 0
+    pos_ground_truth = ground_truth.space.element(pos_ground_truth)
+
+    # Identify negative part of ground_truth
+    neg_ground_truth = ground_truth.copy()
+    neg_ground_truth = neg_ground_truth.asarray()
+    neg_ground_truth[neg_ground_truth > 0] = 0
+    neg_ground_truth = ground_truth.space.element(neg_ground_truth)
+
+    # Volume of space
+    vol = data.space.domain.volume
+
+    # Mean value of data
+    data_mean = (1 / vol) * (l1_norm(pos_data) -
+                             l1_norm(neg_data))
+
+    # Mean value of ground_truth
+    ground_truth_mean = (1 / vol) * (l1_norm(pos_ground_truth) -
+                                     l1_norm(neg_ground_truth))
+
+    fom = np.abs((data_mean - ground_truth_mean))
+
+    if normalized:
+        fom /= (np.abs(data_mean) + np.abs(ground_truth_mean))
+
+    return fom
+
+
+def standard_deviation_difference(data, ground_truth, mask=None,
+                                  normalized=False):
+    """FOM returning absolute difference in standard deviation between
```

Comment:
You also don't *have to* refer to the input parameters if that helps. But abbreviations are OK I think, if they're obvious.